main.tf

```diff
@@ -120,13 +120,43 @@ resource "aws_db_instance" "db_including_name" {
   storage_encrypted           = "${var.storage_encrypted}"
   storage_type                = "${var.storage_type}"
   username                    = "${var.database_user}"
+  tags                        = "${merge(var.tags, map("Name", format("%s-%s", var.environment, var.name)), map("Env", var.environment))}"
+}
 
-  tags = "${merge(var.tags, map("Name", format("%s-%s", var.environment, var.name)), map("Env", var.environment))}"
+# The database instance itself with read replica
+resource "aws_db_instance" "db_including_name_with_read_replica" {
```

Comment:
You probably don't need this at all since I don't think you need to provide a database name for a read replica since I think it gets it from the main instance (not sure on this though, needs testing?)